playbooks/scenario44_rds_instance.yaml

```diff
@@ -59,34 +62,63 @@
     - name: Get info about choosen type of DB
       opentelekomcloud.cloud.rds_flavor_info:
         datastore: "{{  db_type }}"
-        instance_mode: "ha"
+        instance_mode: "single"
       register: rds_flavors
 
-    # Print the first flavor in the query
-    - name: debug
+    # Find flavor with smallest RAM and smallest VCPUs
+    - name: Set fact for smallest flavor based on RAM and VCPUs
+      set_fact:
+        smallest_flavor: "{{ rds_flavors.rds_flavors | sort(attribute='ram') | sort(attribute='vcpus') | first }}"
+
+    # Print the smallest flavor details
+    - name: Print smallest flavor
+      ansible.builtin.debug:
+        msg: "{{ smallest_flavor.name }}"
+    
+    # Set storage type
+    - name: Set storage type based on RDS flavor spec code structure
+      set_fact:
+        volume_type: "{{ 'CLOUDSSD' if smallest_flavor.name.split('.')[2] in ['x1', 'n1'] else 'ULTRAHIGH' }}"
```

Comment:
there's no API providing relation between flavor and volume type?

Reply:
There is an API to query STORAGE TYPE and within Storage type , there is parameter "support_compute_group_type".
And this could be "normal" for c2, s1,m1 ...  in case of ULTRAHIGH
or this could be "general" for  n1 or "dedicated" for x1  in case of CLOUDSSD or ESSD

But I can't found such possibilities via ANSIBLE collections.